File: Modules/Common/Resources/SIE.Web.Resources/CalendarSchemes/Commands/CalendarWeekEditCommand.js
SIE.defineCommand('SIE.Web.Resources.CalendarSchemes.Commands.CalendarWeekEditCommand', {
    extend: 'SIE.cmd.Edit',
    meta: { text: "修改", group: "edit" },

    canExecute: function (view) {
        this.selectedItems = view.getSelection();
        if (this.selectedItems.length > 1) {
            return false;
        }
        var entity = view.getCurrent();
        var parent = view.getParent().getCurrent();
        if (entity == null || parent == null || parent.data.ActiveDate > new Date()) {
            return false;
        }
        return true;
    }
});

File: Modules/Common/Resources/SIE.Wpf.Resources/CalendarSchemes/Commands/CalendarWeekEditCommand.cs
using SIE.Resources.CalendarSchemes;
using SIE.Wpf.Command;
using System;

namespace SIE.Wpf.Calendars.CalendarSchemes.Commands
{
    /// <summary>
    /// 修改
    /// </summary>
    [Command(ImageName = "EditEntity", Label = "修改", GroupType = CommandGroupType.Edit)]
    public class CalendarWeekEditCommand : ListEditCommand
    {
        /// <summary>
        /// 是否可执行
        /// </summary>
        /// <param name="view">视图</param>
        /// <returns>bool</returns>
        public override bool CanExecute(ListLogicalView view)
        {
            return view.Current != null && view.Parent.Current != null && (view.Parent.Current as CalendarSchemeWeek).ActiveDate > DateTime.Today;
        }
    }
}
